components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt

```diff
@@ -224,6 +224,125 @@ class FirefoxAccount(handle: FxaHandle, persistCallback: PersistCallback?) : Aut
         }.getAndConsumeRustString()
     }
 
+    /**
+     * Update the push subscription details for the current device.
+     * This needs should be called every time a push subscription is modified or expires.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @param endpoint Push callback URL
+     * @param endpoint Public key used to encrypt push payloads
+     * @param endpoint Auth key used to encrypt push payloads
+     */
+    fun setDevicePushSubscription(endpoint: String, publicKey: String, authKey: String) {
+        rustCall { e ->
+            LibFxAFFI.INSTANCE.fxa_set_push_subscription(this.handle.get(), endpoint, publicKey, authKey, e)
+        }
+    }
+
+    /**
+     * Update the display name (as shown in the FxA device manager, or the Send Tab target list)
+     * for the current device.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @param displayName The current device display name
+     */
+    fun setDeviceDisplayName(displayName: String) {
+        rustCall { e ->
+            LibFxAFFI.INSTANCE.fxa_set_display_name(this.handle.get(), displayName, e)
+        }
+    }
+
+    /**
+     * Retrieves the list of the connected devices in the current account, including the current one.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     */
+    fun getDevices(): Array<Device> {
+        val devicesBuffer = rustCall { e ->
+            LibFxAFFI.INSTANCE.fxa_get_devices(this.handle.get(), e)
+        }
+        try {
+            val devices = MsgTypes.Devices.parseFrom(devicesBuffer.asCodedInputStream()!!)
+            return Device.fromCollectionMessage(devices)
+        } finally {
+            LibFxAFFI.INSTANCE.fxa_bytebuffer_free(devicesBuffer)
+        }
+    }
+
+    /**
+     * Retrieves any pending commands for the current device.
+     * This should be called semi-regularly as the main method of commands delivery (push)
+     * can sometimes be unreliable on mobile devices.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @return A collection of [AccountEvent] that should be handled by the caller.
+     */
+    fun pollDeviceCommands(): Array<AccountEvent> {
+        val eventsBuffer = rustCall { e ->
+            LibFxAFFI.INSTANCE.fxa_poll_device_commands(this.handle.get(), e)
+        }
+        this.tryPersistState()
```

Comment:
Is it accurate that, if the application were to crash right here, the fetched commands would be lost and never handled?  I believe that matches what desktop does, and it seems fine. May be worth a comment for future readers, in case we decide to revisit this in future?